pkg/engine/loadbalancers.go

```diff
@@ -236,6 +236,62 @@ func CreateLoadBalancer(prop *api.Properties, isVMSS bool) LoadBalancerARM {
 	return loadBalancer
 }
 
+func CreateAgentLoadBalancer(prop *api.Properties, isVMSS bool) LoadBalancerARM {
+
+	loadBalancer := LoadBalancerARM{
+		ARMResource: ARMResource{
+			APIVersion: "[variables('apiVersionNetwork')]",
+			DependsOn: []string{
+				"[concat('Microsoft.Network/publicIPAddresses/', variables('agentPublicIPAddressName'))]",
+			},
+		},
+		LoadBalancer: network.LoadBalancer{
+			Location: to.StringPtr("[variables('location')]"),
+			Name:     to.StringPtr("[variables('agentLbName')]"),
+			LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
+				BackendAddressPools: &[]network.BackendAddressPool{
+					{
+						Name: to.StringPtr("[variables('agentLbBackendPoolName')]"),
+					},
+				},
+				FrontendIPConfigurations: &[]network.FrontendIPConfiguration{
+					{
+						Name: to.StringPtr("[variables('agentLbIPConfigName')]"),
+						FrontendIPConfigurationPropertiesFormat: &network.FrontendIPConfigurationPropertiesFormat{
+							PublicIPAddress: &network.PublicIPAddress{
```

Comment:
just FYI - AKS has run into the requirement that customer wants to bring their own public IP prefix instead of ip address, not sure if AKS-E users would ask for that or not later

Reply:
Yea I think there is an existing issue specifically for Public IP Prefix. That should be addressed in a separate PR.